### PR TITLE
add a wrapper to gdb which adds gcc (or other paths) as safe path

### LINKED ISSUE #10427: gdb auto-load safe-path should include ${pkgs.gcc}
by default, if you are debugging with gdb, stl containers like std::vector can't be pretty-printed using the pretty-printers that ship with gcc, because gdb doesn't know that it can trust things in /nix/store/*gcc*

the safe-path should be configured at compile time or with a wrapper to include this path

### PR DESCRIPTION
**Pull request by fkz**
This fixes #10427. Though I'm not sure if defaulting to adding gcc is the right way to do it.

### CODE REVIEW

**Comment by dezgeg** [comment]
I think this would make gdb manpages not show up in the Nix profile path.

**Comment by fkz** [comment]
you're right, I thought the propagatedBuildInput would magically do the trick when in fact we need propagatedUserEnvPkgs and then you also have to run the default builder...
Anyway, now it seems to work. Alternatively, we could also create a symlink tree with buildEnv

**Comment by fkz** [comment]
hm, travis seems only to fail because 'no space left on device'. Not sure why it rebuilds so much, maybe a lot of qt/kde depends on gdb.
So, this works for me now, but thinking a little more about it I think it would be better to add some hook that adds all (or some?) packages of the current build environment to the safe gdb path. The problem is that, afaik gdb doesn't read any environment variables to add safe paths. So we have to add a "-x initscript" flag to the gdb call which this wrapper does, but I don't see how we can do this depending on hooks. An alternative might be to patch gdb to include an init script that reads the safe paths from some environment variable and then just change them in the hook. I'm also not sure how this works in the "nix-env -iA gdb"/environment.systemPackages=[gdb] case.
I'm also not completely sure of the security implications of adding all buildInputs to the safe paths, but I think you trust your buildInputs anyway.
Any comments?

**Comment by rasendubi** [comment]
(triage) @fkz any update?

**Comment by schmittlauch** [comment]
Just ran into #10427 again, this still needs fixing.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -5897,13 +5897,15 @@ let
     ruby = ruby_2_0_0;
   };
 
-  gdb = callPackage ../development/tools/misc/gdb {
+  gdbWithoutWrapper = callPackage ../development/tools/misc/gdb {
```
Comment (Mic92) [inline pkgs/top-level/all-packages.nix]: Commonly we use the `gdb-unwrapped` prefix instead.

**Comment by Mic92** [comment]
A quick hack if you trust your nix store would be:

```
# ~/.gdbinit
set auto-load safe-path /nix
```

**Comment by Tmplt** [comment]
Are there plans to merge this? I can reproduce #10427 in 18.09, stable at aabc61049c011b40ef15c354d7d84c840d6b6a6d.

**Comment by mmahut** [comment]
Any update on this pull request?

**Comment by mmahut** [comment]
Closing due to lack of activity, feel free to reopen this if needed.

**Comment by tmplt** [comment]
Superseded by https://github.com/NixOS/nixpkgs/pull/73574.

**Pull request closed without merging**